### DISCUSSION TITLE
Thanos Receiver - why does it need consistent hashing?

### DISCUSSION THREAD

**Discussion by oronsh** [thanos-io/thanos#4659] [Questions & Answers]
Hello,
I'm trying to understand why thanos receive uses a hashring as its load balancer rather than a simple round robin?
Thanks a lot!

**Comment by bill3tt** [comment]
Hey @oronsh - with a simple round robin load balancer you would end up writing a subset of samples for every metric to every Thanos receive instance! This would impose huge overheads on the index & memory requirements of each receive instance.

We use a hashring so that we can achieve [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing) of the data. This means that if an element is added or removed from the hashring, the minimal number of elements possible are re-assigned across receive instances.

Hope that makes sense!
